src/main/java/uk/gov/hmcts/reform/blobrouter/services/SasTokenGeneratorService.java

```diff
@@ -58,9 +59,9 @@ private BlobServiceSasSignatureValues getBlobServiceSasSignatureValues(String se
     }
 
     private StorageConfig getConfigForService(String serviceName) {
-        if (!serviceConfiguration.getStorageConfig().containsKey(serviceName)) {
-            throw new ServiceConfigNotFoundException("No service configuration found for " + serviceName);
-        }
-        return serviceConfiguration.getStorageConfig().get(serviceName);
+        return Optional
+            .ofNullable(serviceConfiguration.getStorageConfig().get(serviceName))
+            .filter(StorageConfig::isEnabled)
+            .orElseThrow(() -> new ServiceConfigNotFoundException("No service configuration found for " + serviceName));
```

Comment:
Do we want to throw same exception for services which are disabled as well? Exception gets translated into `BAD_REQUEST` btw.

If we do, introducing `Optional` here perhaps is okay, otherwise we could use map functionality alone: `map.computeIfAbsent`